confdgnmi/src/confd_gnmi_api_adapter.py

```diff
@@ -199,7 +354,8 @@ def get_maapi_save_string(self, path_str, save_flags):
         return save_str
 
     def get_xml_paths(self, elem, orig_path_str, parent_path_cand=""):
-        log.debug("==> elem=%s", elem)
+        log.debug("==> elem=%s orig_path_str=%s parent_path_cand=%s", elem,
```

Comment:
You have entry/exit logs like this pretty much everywhere; maybe time for a function/method decorator? That would have the advantage that you can log non-local exits too.

Reply:
Yes, decorator might be fine. I like it better this way as it is more flexible for development/debugging.